contents/node/nodejs.md

```diff
@@ -0,0 +1,203 @@
+
+### Others
+
+For more information about `package.json`, refer to [npm official documentation](https://docs.npmjs.com/files/package.json).
+
+# Node module system
+
+Let's move on to the module system which plays a central rode in Node. This system allows you to include other Javascript files and thus makes it easy to reuse external libraries and organize your code into separate parts with limited responsibilities.
+
+Node comes bundled with useful [core modules](https://nodejs.org/api/) such as the `fs` (file system) module which includes classes, methods and events to deal with file I/O operations and the `http` module which helps Node to transfer data over HTTP.
+
+There are also useful and well-tested non-core modules maintained by the community and external developers such as *Bluebird* for promises. Before importing them, remember to install them and include them in your `package.json` dependencies by following the instructions from the [npm section](#node-package-manager-npm).
+
+Importing modules is simple - just use the `require()` function and provide the module identifier or the file path.
+```js
+const http = require('http'); // import a core module
+const Promise = require('bluebird'); // import a non-core module
+```
+
+Node will first check if the module identifier passed to `require` is a core module or a relative path. If so, it will return the core module or the object which references the value of `module.exports` in the specified file path respectively. Otherwise, Node will attempt to load the module from the `node_modules` folder in the parent directory of the current module.
+
+### Organizing code
+
+We can also create separate files and modules in our codebase such that each module is focused on a single functionality. This makes your code more maintainable and testable.
+
+To export the `Parser` constructor in `parser.js`,
+```js
+function Parser(options) {
+    this._options = options || {};
+}
+
+Parser.prototype.parse = function (content) {
+  // insert code here
+}
+module.exports = Parser; // override the exports object
+```
+
+The Parser constructor will then be returned whenever the `require` function is used to include the `Parser` module. It can be used to create Parser objects in other files in the project, for example,
+```js
+const parser = require('../parser') // import content from parser.js based on relative file path. The js extension is assumed and can be excluded.
+
+const content = 'some content';
+const newParser = new Parser();
+newParser.parse(content);
+```
+
+To better understand how to use `module.exports`, checkout [Tendai Mutunhire's article](http://stackabuse.com/how-to-use-module-exports-in-node-js/).
+
+# Writing a Hello World Node.js server
```

Comment:
This example serves to motivate the reader to adopt node, by showing how easy it is to use. If possible, move this section higher up (under the `Getting started` section?). It will motivate the reader to keep reading the remainder of the chapter.